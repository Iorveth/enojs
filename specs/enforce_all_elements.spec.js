const { parse } = require('../eno.js');

const sample = `
fieldset:
present = yes

# section

fieldset in section:
present = yes
`;

describe('enforceAllElements', () => {
  it('is not active by default', () => {
    const doc = parse(sample);
    const section = doc.section('section');
    const result = {
      fieldsetEntry: doc.fieldset('fieldset').entry('missing'),
      field: doc.field('missing'),
      section: {
        fieldsetEntry: section.fieldset('fieldset in section').entry('missing'),
        field: section.field('missing')
      }
    };

    const expected = {
      fieldsetEntry: null,
      field: null,
      section: {
        fieldsetEntry: null,
        field: null
      }
    };

    expect(result).toEqual(expected);
  });

  it('is enforced when requested', () => {
    const doc = parse(sample);

    doc.enforceAllElements();

    const section = doc.section('section');

    expect(() => doc.fieldset('fieldset').entry('missing')).toThrowErrorMatchingSnapshot();
    expect(() => doc.field('missing')).toThrowErrorMatchingSnapshot();
    expect(() => section.fieldset('fieldset in section').entry('missing')).toThrowErrorMatchingSnapshot();
    expect(() => section.field('missing')).toThrowErrorMatchingSnapshot();
  });
});
