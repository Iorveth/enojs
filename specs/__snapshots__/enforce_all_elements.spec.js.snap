// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enforceAllElements is enforced when requested 1`] = `
"The fieldset entry 'missing' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
      1 | 
 >    2 | fieldset:
 *    3 | present = yes
      4 | 
      5 | # section
   ...
"
`;

exports[`enforceAllElements is enforced when requested 2`] = `
"The field 'missing' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
 *    1 | 
 *    2 | fieldset:
 *    3 | present = yes
 *    4 | 
 *    5 | # section
 *    6 | 
 *    7 | fieldset in section:
 *    8 | present = yes
 *    9 | 
"
`;

exports[`enforceAllElements is enforced when requested 3`] = `
"The fieldset entry 'missing' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
   ...
      5 | # section
      6 | 
 >    7 | fieldset in section:
 *    8 | present = yes
      9 | 
"
`;

exports[`enforceAllElements is enforced when requested 4`] = `
"The field 'missing' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
   ...
      3 | present = yes
      4 | 
 >    5 | # section
 *    6 | 
 *    7 | fieldset in section:
 *    8 | present = yes
 *    9 | 
"
`;
